src/shared/components/ZendeskWrapper/ZendeskWrapper.tsx

```diff
@@ -0,0 +1,72 @@
+import { get } from 'lodash-es';
+import React, { FunctionComponent } from 'react';
+import Zendesk from 'react-zendesk';
+
+import { getEnv } from '../../helpers';
+
+declare const ResizeObserver: any;
+
+const ZendeskWrapper: FunctionComponent = () => {
```

Comment:
Ik zou die `onLoaded` volledig uit de template trekken en wat opsplitsen.

```
const DEFAULT_FOOTER_HEIGHT = 100;

const ZendeskWrapper: FunctionComponent = () => {
	const getFooterHeight = () => get(
		document.querySelector('.c-global-footer'),
		'clientHeight',
		DEFAULT_FOOTER_HEIGHT
	);

	const updateMarginBottom = (widget: HTMLIFrameElement | null) => {
		if (widget) {
			const newWidget = { ...widget };
			const { scrollHeight, screenHeight, scrollTop } = document.documentElement;

			if (
				scrollHeight - screenHeight - scrollTop <
				footerHeight + zendeskMarginBottom
			) {
				newWidget.style.marginBottom = `${footerHeight +
					zendeskMarginBottom -
					(scrollHeight - screenHeight - scrollTop)}px`;
			} else {
				newWidget.style.marginBottom = '10px';
			}

			return newWidget;
		}
	}

	const onResize = (widget: HTMLIFrameElement | null) => {
		updateFooterHeight();
		updateMargin(widget);
	}

	const getZendeskWidget = () => {
		widget = document.querySelector('iframe#launcher');

		if (!widget) {
			setTimeout(getZendeskWidget, 100);
		}

		onResize(widget);
	};

	const onLoaded = () => {
		// Change bottom margin of the Zendesk widget to avoid overlap with footer
		let widget: HTMLIFrameElement | null;
		let footerHeight: number = DEFAULT_FOOTER_HEIGHT;
		const zendeskMarginBottom: number = 10;

		document.addEventListener('scroll', updateMargin);
		document.addEventListener('resize', () => onResize(widget));

		resizeObserver.observe(document.body);

		getZendeskWidget();
	}

	return (
		<Zendesk
			zendeskKey={getEnv('ZENDESK_KEY') as string}
			onLoaded={onLoaded}
		/>
	);
};
```